sql_composer/pg/pg_translator.py

```
@@ -0,0 +1,591 @@
+        if query_criteria.page:
+            query_criteria_as_sql += f"\n{self.page_criteria_to_sql(query_criteria.page)}"
+
+        return query_criteria_as_sql, params
+
+    def where_to_sql_with_params(self, where: Where, column: Column) -> Tuple[str, List[Any]]:
+        """
+        Generate parameterized WHERE clause with extracted parameters.
+        Returns a tuple of (SQL with %s placeholders, parameters list).
+        """
+        if len(where.values) == 0:
+            raise ValueError(f"Operator {where.op} requires at least 1 value, got 0")
```

Ruff (0.14.7)

Avoid specifying long messages outside the exception class

(TRY003)